core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/WorkflowActor.scala

```diff
@@ -129,7 +129,18 @@ abstract class WorkflowActor(logStorageType: String, val actorId: ActorVirtualId
 
   def receiveDeadLetterMessage: Receive = {
     case MessageBecomesDeadLetter(msg) =>
-      actorRefMappingService.removeActorRef(msg.internalMessage.channel.from)
+      val dest = msg.internalMessage.channel.to
+      if (dest == actorId) {
+        actorService.scheduleOnce(
+          100.millis,
+          () => {
+            logger.warn(s"sending message to self failed, retry sending $msg to self directly.")
+            self ! msg
+          }
+        )
+      } else {
+        actorRefMappingService.removeActorRef(dest)
```

Comment:
why remove dest ("to") where the original code removes "from"?

Reply:
the original code is incorrect.